### DISCUSSION TITLE
Using a step of one job in `if` condition in another job

### DISCUSSION THREAD

**Discussion by kartikdc** [community/community#26536] [Actions]
<p>Hello.</p>
<p>I am trying to understand how to write a workflow. Basically, what I want to do is:</p>
<ol>
<li>Run a job with an action called technote-space/get-diff-action. This saves the diffs in an environment variable called <code>GIT_DIFF</code>.</li>
<li>After this I have five jobs that run tests. I only want to run these <code>if: env.GIT_DIFF</code>, but the environment variable is created in another job so I don’t know how to refer to it.</li>
</ol>
<p>Any idea how I can do this?</p>
<p>This is a reference workflow.yaml file:</p>
<pre><code class="lang-yaml">jobs:
  get_git_diff:
    runs-on: self-hosted
    name: Get git diff
    steps:
      - uses: actions/checkout@v2
        name: Checkout this repository
      - uses: kartikdc/get-diff-action@5.0.0
        name: Get git diff
        id: get_git_diff
        with:
          PATTERNS: |
            !.version
            !README.md
            !docs/**
            !.github/**

  eslint_check:
    if: jobs.get_git_diff.steps.get_git_diff.env.GIT_DIFF
    needs: cancel_previous_actions
    timeout-minutes: 10
    runs-on: [self-hosted]
    name: Run eslint check
    steps:
</code></pre>
<p>Obviously this doesn’t work. Can someone tell me how I can achieve the desired result? Running the diff check in another job would be cool because there are multiple test jobs. Otherwise, I would have to add it as a step in all jobs.</p>
<p>Any help is appreciated. Thanks in advance.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Environment variables do not carry over between jobs. You can use job outputs to make data available from a job to others that have a <code>needs</code> dependency on it, see <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs"><code>jobs.&lt;job_id&gt;.outputs</code></a>.</p>

**Comment by kartikdc** [comment]
<p>Thank you very much Luna</p>
